front-end/src/components/analysis-page/IntentMenu.test.js

```diff
@@ -120,6 +123,151 @@ describe("IntentMenu", () => {
+            props.intents[0].star = true;
+            props.intents[0].flag = true;
+            renderComponent(props);
+            userEvent.click(screen.getByTestId("star-filter"));
+            userEvent.click(screen.getByTestId("flag-filter"));
+            // A button for Intent 0 should display
+            Object.keys(props.intents).forEach((key) => {
+                expect(screen.queryAllByLabelText(`${props.intents[key].question}`).length).toBe(key < 1 ? 1 : 0);
+            });
+            // To get make the filters false for future testing
+            userEvent.click(screen.getByTestId("star-filter"));
+            userEvent.click(screen.getByTestId("flag-filter"));
+        });
+
+        it ("should display two intents when two intents are both starred and flagged", () => {
+            props.intents[0].star = true;
+            props.intents[0].flag = true;
+            props.intents[1].star = true;
+            props.intents[1].flag = true;
+            renderComponent(props);
+            userEvent.click(screen.getByTestId("star-filter"));
+            userEvent.click(screen.getByTestId("flag-filter"));
+            // A button for each of Intents 0-1 should display
+            Object.keys(props.intents).forEach((key) => {
+                expect(screen.queryAllByLabelText(`${props.intents[key].question}`).length).toBe(key < 2 ? 1 : 0);
+            });
+            // To get make the filters false for future testing
+            userEvent.click(screen.getByTestId("star-filter"));
+            userEvent.click(screen.getByTestId("flag-filter"));
+        });
+
+        it ("should display two intents when two intents are both starred and flagged and menu is closed and opened", () => {
+            props.intents[0].star = true;
+            props.intents[0].flag = true;
+            props.intents[1].star = true;
+            props.intents[1].flag = true;
+            renderComponent(props);
+            userEvent.click(screen.getByTestId("star-filter"));
+            userEvent.click(screen.getByTestId("flag-filter"));
+            // A button for each of Intents 0-1 should display
+            Object.keys(props.intents).forEach((key) => {
+                expect(screen.queryAllByLabelText(`${props.intents[key].question}`).length).toBe(key < 2 ? 1 : 0);
+            });
+            // Close intent menu
+            userEvent.click(screen.getByLabelText("Close intent menu"));
+            // Button amount stays the same even when closed
+            Object.keys(props.intents).forEach((key) => {
+                expect(screen.queryAllByLabelText(`${props.intents[key].question}`).length).toBe(key < 2 ? 1 : 0);
+            });
+            // To get make the filters false for future testing
+            userEvent.click(screen.getByTestId("star-filter"));
+            userEvent.click(screen.getByTestId("flag-filter"));
+        });
+    });
```

Comment:
extraneous tests

Reply:
I'll get rid of most of the extra tests but I might keep the one for after intent menu is closed that it will stay the same because that was a bug that Kaylee ran into when she was first testing.